dot/sync/message_test.go

```diff
@@ -0,0 +1,439 @@
+		chainProcessor ChainProcessor
+		network        Network
+	}
+	type args struct {
+		ancestor         common.Hash
+		descendant       *common.Hash
+		descendantNumber uint
+	}
+	tests := []struct {
+		name          string
+		fields        fields
+		args          args
+		want          common.Hash
+		expectedError error
+	}{
+		{
+			name: "nil descendant",
+			fields: fields{
+				blockState: newMockBlockStateForMessageTest(ctrl)},
+			args: args{ancestor: common.Hash{}, descendant: nil, descendantNumber: 1},
+		},
+		{
+			name: "not nil descendant",
+			fields: fields{
+				blockState: newMockBlockStateForMessageTest(ctrl)},
+			args: args{ancestor: common.Hash{0}, descendant: &common.Hash{1, 2}, descendantNumber: 1},
+			want: common.Hash{1, 2},
+		},
+		{
+			name: "descendant greater than header",
+			fields: fields{
+				blockState: newMockBlockStateForMessageTest(ctrl)},
+			args:          args{ancestor: common.Hash{2}, descendant: &common.Hash{1, 2}, descendantNumber: 1},
+			want:          common.Hash{},
+			expectedError: errors.New("invalid request, descendant number 2 is higher than ancestor 1"),
+		},
+	}
+	for _, tt := range tests {
+		tt := tt
+		t.Run(tt.name, func(t *testing.T) {
+			t.Parallel()
+			s := &Service{
+				blockState:     tt.fields.blockState,
+				chainSync:      tt.fields.chainSync,
+				chainProcessor: tt.fields.chainProcessor,
+				network:        tt.fields.network,
+			}
+			got, err := s.checkOrGetDescendantHash(tt.args.ancestor, tt.args.descendant, tt.args.descendantNumber)
+			if tt.expectedError != nil {
+				assert.EqualError(t, err, tt.expectedError.Error())
+			}
+			if !reflect.DeepEqual(got, tt.want) {
+				t.Errorf("checkOrGetDescendantHash() got = %v, want %v", got, tt.want)
+			}
+		})
+	}
+}
+
+func TestService_getBlockData(t *testing.T) {
+	t.Parallel()
+
+	ctrl := gomock.NewController(t)
+
+	mockBlockState := NewMockBlockState(ctrl)
+	mockBlockState.EXPECT().GetHeader(common.Hash{}).Return(nil, errors.New("empty hash"))
+	mockBlockState.EXPECT().GetHeader(common.Hash{1}).Return(&types.Header{
+		Number: 2,
+	}, nil)
+	mockBlockState.EXPECT().GetBlockBody(common.Hash{}).Return(nil, errors.New("empty hash"))
+	mockBlockState.EXPECT().GetBlockBody(common.Hash{1}).Return(&types.Body{[]byte{1}}, nil)
+	mockBlockState.EXPECT().GetReceipt(common.Hash{1}).Return([]byte{1}, nil)
+	mockBlockState.EXPECT().GetMessageQueue(common.Hash{2}).Return([]byte{2}, nil)
+	mockBlockState.EXPECT().GetJustification(common.Hash{3}).Return([]byte{3}, nil)
+
+	type fields struct {
+		blockState BlockState
+	}
+	type args struct {
+		hash          common.Hash
+		requestedData byte
+	}
+	tests := []struct {
+		name   string
+		fields fields
+		args   args
+		want   *types.BlockData
+		err    error
+	}{
+		{
+			name: "requestedData 0",
+			args: args{
+				hash:          common.Hash{},
+				requestedData: 0,
+			},
+			want: &types.BlockData{},
+		},
+		{
+			name: "requestedData RequestedDataHeader, error",
```

Comment:
nit / FYI commas don't render well when running the tests since it replaces spaces with `_`

```suggestion
			name: "requestedData RequestedDataHeader error",
```